operator/internal/controller/redpanda/redpanda_decommission_controller.go

```diff
@@ -360,23 +368,36 @@ func (r *DecommissionReconciler) reconcileDecommission(ctx context.Context, log
 			nodesDownMap.Add(node)
 		}
 
+		bl, err := adminAPI.Brokers(ctx)
+		if err != nil {
+			return ctrl.Result{}, fmt.Errorf("could not get brokers: %w", err)
+		}
+
+		addresses := map[string]any{}
+		for _, b := range bl {
+			addresses[b.InternalRPCAddress] = struct{}{}
```

Comment:
Is this logic pulled from somewhere else? I don't think I've seen use use `InternalRPCAddress` as a deduplication method before. Is there a precedence for doing so?